sky/provision/kubernetes/utils.py

```diff
@@ -1121,6 +1121,19 @@ def _merge_dicts(source, destination):
                     assert len(value) == 1, \
                         f'Expected only one container, found {value}'
                     _merge_dicts(value[0], destination[key][0])
+                elif key in ['volumes', 'volumeMounts']:
+                    # If the key is 'volumes' or 'volumeMounts', we search for
+                    # item with the same name and merge it.
+                    for new_volume in value:
+                        new_volume_name = new_volume.get('name')
+                        if new_volume_name is not None:
+                            destination_volume = next(
+                                (v for v in destination[key]
+                                 if v.get('name') == new_volume_name), None)
```

Comment:
nit: using filter might be easier to read. not feeling strongly, please feel free to ignore.
```suggestion
                            destination_volume = next(filter(lambda v: v.get('name') == new_volume_name, destination[key]), None)
```

Reply:
the generator might be [more pythonic](https://chat.openai.com/c/20cad2e1-b3ef-48ec-95ef-3b67e1adc23b) :)